mysensors/mysensors.py

```diff
@@ -280,14 +297,15 @@ def _perform_file_action(self, filename, action):
             raise Exception('Unsupported file type %s' % ext[1:])
         func(filename)
 
-    def alert(self, nid):
+    def alert(self, msg):
         """Tell anyone who wants to know that a sensor was updated.
 
         Also save sensors if persistence is enabled.
         """
         if self.event_callback is not None:
             try:
-                self.event_callback('sensor_update', nid)
+                setattr(msg, 'gateway', self)
```

Comment:
My preference is that Message class would be updated to officially include a reference to the related gateway. Setting the attribute for the callback was a way to limit the scope of my proposed changes.

The issue is that the gateway constructor takes the event handler as a constructor argument. If the handler constructor needs a reference to the gateway (for resolving gateway-specific constants, for example) there is a messy circular dependency. This would be a scenario where the user has created a stateful event callback (a callable object or a bound function on an object with state) versus a simple function callback. There are workarounds. With the current implementation, one could not use the event_handler constructor argument and set the event_handler on the gateway after it is created. Or one could set the gateway on an event handler object after the gateway is created. Or the handler could access the gateway as a global variable, but that would prevent a generic handler from being used with multiple gateways. It seemed cleaner to me to give the callback access to the gateway that processed the incoming message. However, if you prefer one of the workarounds instead I can remove the related code from the PR.